neptune/hierarchy_test.go

```diff
@@ -65,6 +67,103 @@ func TestNeptuneDB_GetCodesWithData(t *testing.T) {
 	})
 }
 
+func TestNeptuneDB_HierarchyExists(t *testing.T) {
+
+	vertex := internal.MakeHierarchyVertex("vertex-label", "code", "label", 1, true)
+
+	Convey("Given a mocked neptune DB that returns a single hierarchy node", t, func() {
+
+		poolMock := &internal.NeptunePoolMock{GetFunc: func(query string, bindings map[string]string, rebindings map[string]string) (vertices []graphson.Vertex, err error) {
+			return []graphson.Vertex{vertex}, nil
+		}}
+		db := mockDB(poolMock)
+
+		Convey("When HierarchyExists is called", func() {
+			hierarchyExists, err := db.HierarchyExists(ctx, testInstanceID, testDimensionName)
+
+			Convey("Then the expected query is sent to Neptune", func() {
+
+				expectedQuery := fmt.Sprintf(query.HierarchyExists, testInstanceID, testDimensionName)
+				So(len(poolMock.GetCalls()), ShouldEqual, 1)
+				So(poolMock.GetCalls()[0].Query, ShouldEqual, expectedQuery)
+			})
+
+			Convey("Then the return value is true", func() {
+				So(hierarchyExists, ShouldBeTrue)
+				So(err, ShouldBeNil)
+			})
+		})
+	})
+
+	Convey("Given a mocked neptune DB that returns multiple hierarchy nodes", t, func() {
+
+		poolMock := &internal.NeptunePoolMock{GetFunc: func(query string, bindings map[string]string, rebindings map[string]string) (vertices []graphson.Vertex, err error) {
+			return []graphson.Vertex{vertex, vertex}, nil
+		}}
+		db := mockDB(poolMock)
+
+		Convey("When HierarchyExists is called", func() {
+			_, err := db.HierarchyExists(ctx, testInstanceID, testDimensionName)
```

Comment:
Probably should assign and check the `bool` return as well as it is an output value of the function being tested.